de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SessionUtils.java

```diff
@@ -0,0 +1,82 @@
+package de.fu_berlin.inf.dpp.session;
+
+import de.fu_berlin.inf.dpp.SarosPluginContext;
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.filesystem.IResource;
+
+import org.picocontainer.annotations.Inject;
+
+/**
+ * A utility class to provide easy access to functionality needing a session
+ * object.
+ */
+public class SessionUtils {
+    private static volatile ISarosSession currentSarosSession;
+
+    @Inject
+    private static ISarosSessionManager sarosSessionManager;
+
+    static {
+        SarosPluginContext.initComponent(new SessionUtils());
+
+        sarosSessionManager
+            .addSessionLifecycleListener(new NullSessionLifecycleListener() {
+                @Override
+                public void sessionStarted(ISarosSession session) {
+
+                    setSession(session);
+                }
+
+                @Override
+                public void sessionEnded(ISarosSession session,
+                    SessionEndReason reason) {
+
+                    resetSession();
+                }
+            });
+    }
+
+    private static void setSession(ISarosSession newSarosSession) {
+        currentSarosSession = newSarosSession;
+    }
+
+    private static void resetSession() {
+        currentSarosSession = null;
+    }
+
+    /**
+     * Returns whether the resource represented by the given SPath is shared.
+     *
+     * @param resource the resource to check
+     * @return <code>true</code> if a session is currently running and the given
+     * resource exists and is shared, <code>false</code> otherwise
+     */
+    public static boolean isShared(IResource resource) {
+
+        assert resource != null : "The given IResource must not be null.";
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this assert with a proper check. ([squid:S4274](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS4274))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=293&issues=AWbRiJzSSgN4uCclDZzw&open=AWbRiJzSSgN4uCclDZzw)

[ISSUE_KEY]: AWbRiJzSSgN4uCclDZzw